PWGHF/HFL/Tasks/taskSingleMuon.cxx

```diff
@@ -120,46 +122,51 @@ struct HfTaskSingleMuonSelection {
     registry.fill(HIST("hVtxZ"), collision.posZ());
 
     for (const auto& muon : muons) {
-      if (muon.trackType() == trkType) {
-        const auto eta(muon.eta()), pDca(muon.pDca()), rAbs(muon.rAtAbsorberEnd());
-        const auto dcaXY(std::sqrt(std::pow(muon.fwdDcaX(), 2.) + std::pow(muon.fwdDcaY(), 2.)));
-        const auto pt(muon.pt());
-        const auto charge(muon.sign());
-        const auto chi2(muon.chi2MatchMCHMFT());
-
-        auto trkMFT = muon.template matchMFTTrack_as<MFTTracksExtra>();
-        if (reduceAmbMFT && trkMFT.has_ambMFTtrack()) {
-          continue;
-        }
-        // histograms before the acceptance cuts
-        registry.fill(HIST("hMuBeforeCuts"), pt, eta, dcaXY, pDca, charge, chi2);
-        if (muon.matchMCHTrackId() > 0.) {
-          auto muonType3 = muon.template matchMCHTrack_as<MyMuons>();
-          registry.fill(HIST("h3DeltaPtBeforeCuts"), pt, eta, muonType3.pt() - pt);
-        }
-        // standard muon acceptance cuts
-        if ((eta >= etaUp) || (eta < etaLow)) {
-          continue;
-        }
-        if ((rAbs >= rAbsUp) || (rAbs < rAbsLow)) {
-          continue;
-        }
-        if ((rAbs < rAbsMid) && (pDca >= pDcaLow)) {
-          continue;
-        }
-        if ((rAbs >= rAbsMid) && (pDca >= pDcaUp)) {
-          continue;
-        }
-        if ((muon.chi2() >= 1e6) || (muon.chi2() < 0.)) {
-          continue;
-        }
-        // histograms after acceptance cuts
-        registry.fill(HIST("hMuAfterCuts"), pt, eta, dcaXY, pDca, charge, chi2);
-        registry.fill(HIST("h2DCA"), muon.fwdDcaX(), muon.fwdDcaY());
-        if (muon.matchMCHTrackId() > 0.) {
-          auto muonType3 = muon.template matchMCHTrack_as<MyMuons>();
-          registry.fill(HIST("h3DeltaPtAfterCuts"), pt, eta, muonType3.pt() - pt);
-        }
+      if (muon.trackType() != trkType) {
+        continue;
+      }
+      const auto eta(muon.eta()), pDca(muon.pDca()), rAbs(muon.rAtAbsorberEnd());
+      //        const auto dcaXY(std::sqrt(std::pow(muon.fwdDcaX(), 2.) + std::pow(muon.fwdDcaY(), 2.)));
+      const auto dcaXY(RecoDecay::sqrtSumOfSquares(muon.fwdDcaX(), muon.fwdDcaY()));
+      const auto pt(muon.pt());
+      const auto charge(muon.sign());
+      const auto chi2(muon.chi2MatchMCHMFT());
+
+      auto trkMFT = muon.template matchMFTTrack_as<MFTTracksExtra>();
+      if (reduceAmbMft && trkMFT.has_ambMftTrack()) {
+        continue;
+      }
+      if (reduceOrphMft && (!reduceAmbMft) && trkMFT.has_ambMftTrack()) {
+        continue;
+      }
+      // histograms before the acceptance cuts
+      registry.fill(HIST("hMuBeforeCuts"), pt, eta, dcaXY, pDca, charge, chi2);
+      if (muon.matchMCHTrackId() > 0.) {
```

Comment:
```suggestion
      if (muon.matchMCHTrackId() > 0) {
```

Reply:
Hi Vit, the format problems have been fixed.